### ISSUE TITLE
Action fails due to fatal "unsafe repository" error

### ISSUE THREAD

**Issue by SaschaMann** [zulip/zulip-archive#81]
Since last night, the archive build in [JuliaCommunity/zulip-archive](https://github.com/JuliaCommunity/zulip-archive) has been failing due to the following error:

```
Run zulip/zulip-archive@7c772a65b1dd1540d2bb64ee6f0b2860ccaf32c0
/usr/bin/docker run --name bcf09c6e0c2241d25423eb0bae2c81957d3c6_e63042 --label 2bcf09 --workdir /github/workspace --rm -e INPUT_ZULIP_ORGANIZATION_URL -e INPUT_ZULIP_BOT_EMAIL -e INPUT_ZULIP_BOT_KEY -e INPUT_GITHUB_PERSONAL_ACCESS_TOKEN -e INPUT_DELETE_HISTORY -e INPUT_ARCHIVE_BRANCH -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_RETENTION_DAYS -e GITHUB_RUN_ATTEMPT -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_REF_NAME -e GITHUB_REF_PROTECTED -e GITHUB_REF_TYPE -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e GITHUB_ACTION_REPOSITORY -e GITHUB_ACTION_REF -e GITHUB_PATH -e GITHUB_ENV -e GITHUB_STEP_SUMMARY -e RUNNER_OS -e RUNNER_ARCH -e RUNNER_NAME -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/zulip-archive/zulip-archive":"/github/workspace" 2bcf09:c6e0c2241d25423eb0bae2c81957d3c6  "***" "***" "***" "***" "true" "master"
fatal: unsafe repository ('/github/workspace' is owned by someone else)
To add an exception for this directory, call:

	git config --global --add safe.directory /github/workspace
```

[(Example build)](https://github.com/JuliaCommunity/zulip-archive/runs/6006995511?check_suite_focus=true)

I'm not sure where this error occurs or where that git command would have to go.

<details><summary>Runner info</summary>

```
Current runner version: '2.289.2'
Operating System
Virtual Environment
  Environment: ubuntu-20.04
  Version: 20220405.4
  Included Software: https://github.com/actions/virtual-environments/blob/ubuntu20/20220405.4/images/linux/Ubuntu2004-Readme.md
  Image Release: https://github.com/actions/virtual-environments/releases/tag/ubuntu20%2F20220405.4
Virtual Environment Provisioner
  1.0.0.0-main-20220325-1
GITHUB_TOKEN Permissions
  Actions: write
  Checks: write
  Contents: write
  Deployments: write
  Discussions: write
  Issues: write
  Metadata: read
  Packages: write
  Pages: write
  PullRequests: write
  RepositoryProjects: write
  SecurityEvents: write
  Statuses: write
Secret source: Actions
Prepare workflow directory
Prepare all required actions
Getting action download info
Download action repository 'actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f' (SHA:5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f)
Download action repository 'zulip/zulip-archive@7c772a65b1dd1540d2bb64ee6f0b2860ccaf32c0' (SHA:7c772a65b1dd1540d2bb64ee6f0b2860ccaf32c0)
```

</details>

**Comment by rht** [comment]
I saw the errors. I tried https://github.com/rht/zulip-archive/commit/b6a6d585f40d21ebd7a4c4ec8594d23dbb493365, but this didn't fix the problem.

**Comment by SaschaMann** [comment]
Potential workarounds here: https://github.com/actions/checkout/issues/760

**Comment by SaschaMann** [comment]
Fixed in #83
